### PR TITLE
[ref]: SearchSorted: Made ref impl parallel.

### PR DESCRIPTION
**Pull request by pkowalc1**
### Details:
 - Made ref impl parallel. Scaling is done via values space.

### Tickets:
 - *CVS-155707*

### CODE REVIEW

**Comment by pkowalc1** [comment]
Android ARM64 with vcpkg is a known issue and will be fixed after https://github.com/openvinotoolkit/openvino/pull/27171 is merged.

File: src/core/reference/include/openvino/reference/search_sorted.hpp [hunk 1]
```diff
@@ -30,7 +31,13 @@ void search_sorted(const T* sorted,
         };
     }
 
-    for (const Coordinate& values_coord : values_transform) {
+    const size_t size = shape_size(values_shape);
+
+    auto func = [&](size_t i) {
+        auto it = values_transform.begin();
```
Comment (t-jankowski) [inline src/core/reference/include/openvino/reference/search_sorted.hpp:37]: Getting [`begin` iterator](https://github.com/openvinotoolkit/openvino/blob/239466ca5d5333af11ee750260215941737383b3/src/core/reference/src/utils/coordinate_transform.cpp#L24) might not be optimized well. What about obtaining it up front?

Comment (pkowalc1) [reply src/core/reference/include/openvino/reference/search_sorted.hpp:37]: Not sure what you mean here, values_transform is of type CoordinateTransformBasic and it only support access via iterators.

Comment (t-jankowski) [reply src/core/reference/include/openvino/reference/search_sorted.hpp:37]: `values_transform.begin()` is called (not inlined) on every `func(..)` call, but it's return value is the same each time. My idea is to create const begin-iter object outside the closure and just assign its value to changeable `it`. Up to you.

Comment (pkowalc1) [reply src/core/reference/include/openvino/reference/search_sorted.hpp:37]: I think they way it is now is more readable. In terms of optimization, I am pretty sure you won't see any difference so I will leave it as it is.

**Review by praasz — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Pull request merged**